Source/Core/Core/ConfigLoaders/MovieConfigLoader.cpp

```diff
@@ -20,14 +18,19 @@
 #include "Core/Movie.h"
 #include "VideoCommon/VideoConfig.h"
 
+namespace PowerPC
+{
+enum class CPUCore;
+}
+
 namespace ConfigLoaders
 {
 static void LoadFromDTM(Config::Layer* config_layer, Movie::DTMHeader* dtm)
 {
   config_layer->Set(Config::MAIN_CPU_THREAD, dtm->bDualCore);
   config_layer->Set(Config::MAIN_DSP_HLE, dtm->bDSPHLE);
   config_layer->Set(Config::MAIN_FAST_DISC_SPEED, dtm->bFastDiscSpeed);
-  config_layer->Set(Config::MAIN_CPU_CORE, static_cast<int>(dtm->CPUCore));
+  config_layer->Set(Config::MAIN_CPU_CORE, static_cast<PowerPC::CPUCore>(dtm->CPUCore));
```

Comment:
This actually works with just a forward declaration? Nice.
Although, guessing it's ok since the members aren't referenced here, just the type.

Reply:
an enum redeclared like:

```cpp
enum class Name;
// or
enum class Name : <integral type>;
// or
enum Name : <integral type>;
```

is actually an opaque enum declaration, which has a complete type. From [dcl.enum] paragraph 3:

> An *opaque-enum-declaration* is either a redeclaration of an enumeration in the current scope or a declaration of a new enumeration. [ *Note:* An enumeration declared by an *opaque-enum-declaration* has fixed underlying type and is a complete type. The list of enumerators can be provided in a later redeclaration with an enum-specifier. *—end note* ] ...

which makes sense, considering the data size and underlying type of the enum would be known in all of those cases, you just don't have the actual members of the enum itself.